### PR TITLE
Tp banner manager editor ui

### PR DESCRIPTION
**Pull request by TomPretty**
## What does this change?
Implement the majority of the styling for the banner manager UI as per [this card](https://trello.com/c/aLBjiWBk/2093-styling-the-ui-for-banner-manager). It also makes some tiny fixes to the epic manager UI, but for now that remains the same as before. 

## Images
banner-manager-ui

### CODE REVIEW

File: public/src/components/channelManagement/hooks/useValidation.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+const useValidation = (onValidationChanged: (isValid: boolean) => void) => {
+  const [validationStatus, setValidationStatus] = useState<ValidationStatus>(
+    {}
+  );
+
+  const [wasValid, setWasValid] = useState(true);
+
+  useEffect(() => {
+    const isValid = allValid(validationStatus);
+    if (isValid !== wasValid) {
```
Comment (tomrf1) [inline public/src/components/channelManagement/hooks/useValidation.ts:17]: is this extra `wasValid` state for performance reasons? Does it change the behaviour of the app?

File: public/src/components/channelManagement/hooks/useValidation.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      setWasValid(isValid);
+      onValidationChanged(allValid(validationStatus));
+    }
+  }, [validationStatus]);
+
+  const setValidationStatusForField = (field: string, isValid: boolean) => {
+    setValidationStatus({ ...validationStatus, [field]: isValid });
```
Comment (tomrf1) [inline public/src/components/channelManagement/hooks/useValidation.ts:24]: Is it safer to do this?
```
    setValidationStatus(current => ({
      ...current,
      [fieldName]: isValid,
    }));
```

File: public/src/components/channelManagement/createTestDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    if (mode === "COPY" && copiedTestName) {
+      setName(`Copy of ${copiedTestName}`);
+    }
+  }, []);
+
+  const getDuplicateNicknameError = createGetDuplicateError(existingNicknames);
+  const getNicknameError = (value: string) =>
+    getEmptyError(value) || getDuplicateNicknameError(value);
+
+  const [
```
Comment (tomrf1) [inline public/src/components/channelManagement/createTestDialog.tsx:87]: All the state returned by `useValidatableField` appears to be used only by a `TextField`.
Why not have a generic text field component with all the state inside? I think this is pretty much what `EditableTextField` does, though as a class component.

File: public/src/components/channelManagement/hooks/useValidation.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import { useState, useEffect } from "react";
+import { ValidationStatus } from "../helpers/validation";
+
+const allValid = (validationStatus: ValidationStatus) => {
+  return Object.values(validationStatus).every((status) => status);
+};
+
+const useValidation = (onValidationChanged: (isValid: boolean) => void) => {
```
Comment (tomrf1) [inline public/src/components/channelManagement/hooks/useValidation.ts:8]: Let's specify the return type for this function, so it's clear how the rest of the program uses this API:
```
type FieldValidationChange = (fieldName: string, isValid: boolean) => void;
const useValidation = (onValidationChanged: (isValid: boolean) => void): FieldValidationChange => { }...
```

File: public/src/components/channelManagement/hooks/useValidation.ts (hunk 1 above)
Comment (tomrf1) [reply public/src/components/channelManagement/hooks/useValidation.ts:17]: discussed IRL, it was triggering an infinite loop without this

File: public/src/components/channelManagement/hooks/useValidation.ts (hunk 4 above)
Comment (TomPretty) [reply public/src/components/channelManagement/hooks/useValidation.ts]: 👍

File: public/src/components/channelManagement/hooks/useValidation.ts (hunk 2 above)
Comment (TomPretty) [reply public/src/components/channelManagement/hooks/useValidation.ts]: 👍

File: public/src/components/channelManagement/createTestDialog.tsx (hunk 3 above)
Comment (TomPretty) [reply public/src/components/channelManagement/createTestDialog.tsx:87]: Will investigate this now!

Comment (TomPretty) [reply public/src/components/channelManagement/createTestDialog.tsx:87]: After IRL chat: will revisit this and put in the time to get a nice, general purpose solution

**Review by tomrf1 — APPROVED** [review]
brilliant!

**Pull request merged**